src/schema.types.ts

```diff
@@ -267,27 +268,32 @@ export function generateUseCaseProviderParametersFields(
   allProviderSettings: ProviderSettingsRecord,
 ): GraphQLInputFieldConfigMap | undefined {
   // Set to prevent duplicated fields; it's okay if there are conflicting fields since we always expect a string
-  const parameterNames = new Set<string>();
+  const parametersWithProviders = new ArrayMultiMap<string, string>();
 
   // Generate a union of all parameters' names by all configured providers
   for (const providerName of configuredProviders) {
     const providerSettings = allProviderSettings[providerName];
     const parameters = Object.keys(providerSettings.parameters);
-    parameters.forEach((parameterName) => parameterNames.add(parameterName));
+    parameters.forEach((parameterName) =>
+      parametersWithProviders.set(parameterName, providerName),
+    );
   }
 
-  if (parameterNames.size === 0) {
+  if (parametersWithProviders.size === 0) {
     return undefined;
   }
 
   debug(
     'generateUseCaseProviderParametersFields found provider parameters %o',
-    parameterNames,
+    parametersWithProviders,
   );
 
   const fields: GraphQLInputFieldConfigMap = {};
-  for (const parameterName of parameterNames.values()) {
-    fields[parameterName] = { type: GraphQLString };
+  for (const [parameterName, providers] of parametersWithProviders) {
+    fields[parameterName] = {
+      description: `Parameter accepted by ${providers.join(', ')}`,
```

Comment:
I wanted to use [Intl.ListFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/ListFormat) here, sadly it's not supported by TS yet (missing in ES2021 types but supported since Node 12).

Reply:
I think bigger problem is not being able to have description from provider.json, this is nice to have.

Reply:
Agreed, unfortunately at this moment we can't access those via OneSDK.